src/measure_theory/decomposition/lebesgue.lean

```diff
@@ -0,0 +1,540 @@
+begin
+  ext x,
+  simp only [option.mem_def, ennreal.some_eq_coe],
+  split; intro h; rw ← h, symmetry,
+  all_goals {
+    set c := (⨆ (k : ℕ) (hk : k ≤ m + 1), f k a) with hc,
+    set d := (f m.succ a ⊔ ⨆ (k : ℕ) (hk : k ≤ m), f k a) with hd,
+    suffices : c ≤ d ∧ d ≤ c,
+    { change c = d, -- removing this line breaks
+      exact le_antisymm this.1 this.2 },
+    rw [hc, hd],
+    refine ⟨_, _⟩,
+    { refine bsupr_le (λ n hn, _),
+      rcases nat.of_le_succ hn with (h | h),
+      { exact le_sup_of_le_right (le_bsupr n h) },
+      { exact h ▸ le_sup_left } },
+    { refine sup_le _ _,
+      { convert @le_bsupr _ _ _ (λ i, i ≤ m + 1) _ m.succ (le_refl _), refl },
+      { refine bsupr_le (λ n hn, _),
+        have := (le_trans hn (nat.le_succ m)), -- replacing `this` below with the proof breaks
+        exact (le_bsupr n this) } } },
+end
+
+lemma supr_mem_measurable_le
+  (f : ℕ → α → ℝ≥0∞) (hf : ∀ n, f n ∈ measurable_le μ ν) (n : ℕ) :
+  (λ x, ⨆ k (hk : k ≤ n), f k x) ∈ measurable_le μ ν :=
+begin
+  induction n with m hm,
+  { refine ⟨_, _⟩,
+    { simp [(hf 0).1] },
+    { intros A hA, simp [(hf 0).2 A hA] } },
+  { have : (λ (a : α), ⨆ (k : ℕ) (hk : k ≤ m + 1), f k a) =
+      (λ a, f m.succ a ⊔ ⨆ (k : ℕ) (hk : k ≤ m), f k a),
+    { exact funext (λ _, supr_succ_eq_sup _ _ _) },
+    refine ⟨measurable_supr (λ n, measurable.supr_Prop _ (hf n).1), λ A hA, _⟩,
+    rw this, exact (sup_mem_measurable_le (hf m.succ) hm).2 A hA }
+end
+
+lemma supr_mem_measurable_le'
+  (f : ℕ → α → ℝ≥0∞) (hf : ∀ n, f n ∈ measurable_le μ ν) (n : ℕ) :
+  (⨆ k (hk : k ≤ n), f k) ∈ measurable_le μ ν :=
+begin
+  convert supr_mem_measurable_le f hf n,
+  ext, simp
+end
+
+lemma supr_monotone {α : Type*} (f : ℕ → α → ℝ≥0∞) :
+  monotone (λ n x, ⨆ k (hk : k ≤ n), f k x) :=
+begin
+  intros n m hnm x,
+  simp only,
+  refine bsupr_le (λ k hk, _),
+  have : k ≤ m := le_trans hk hnm, -- replacing `this` below with the proof breaks
+  exact le_bsupr k this,
+end
+
+lemma supr_monotone' {α : Type*} (f : ℕ → α → ℝ≥0∞) (x : α) :
+  monotone (λ n, ⨆ k (hk : k ≤ n), f k x) :=
+λ n m hnm, supr_monotone f hnm x
+
+lemma supr_le_le {α : Type*} (f : ℕ → α → ℝ≥0∞) (n k : ℕ) (hk : k ≤ n) :
+  f k ≤ λ x, ⨆ k (hk : k ≤ n), f k x :=
+λ x, le_bsupr k hk
+
+/-- `measurable_le_eval μ ν` is the set of `∫⁻ x, f x ∂μ` for all `f ∈ measurable_le μ ν`. -/
+def measurable_le_eval (μ ν : measure α) : set ℝ≥0∞ :=
+(λ f : α → ℝ≥0∞, ∫⁻ x, f x ∂μ) '' measurable_le μ ν
+
+end lebesgue_decomposition
+
+open lebesgue_decomposition
+
+/-- **The Lebesgue decomposition theorem**: Any pair of finite measures `μ` and `ν`
+`have_lebesgue_decomposition`. That is to say, there exists a measure `ξ` and a measurable function
+`f`, such that `ξ` is mutually singular with respect to `ν` and `μ = ξ + ν.with_density f` -/
+theorem have_lebesgue_decomposition_of_finite_measure
+  (ν μ : measure α) [finite_measure ν] [finite_measure μ] :
+  have_lebesgue_decomposition ν μ :=
```

Comment:
In the docstring, and in all the prerequisites, you speak of `have_lebesgue_decomposition μ ν`, but there you have exchanged the role of the two measures. Can you switch back to have something uniform all over the file?